### PR TITLE
Allow customization of error message when JSON error handler failed

### PR DESCRIPTION
**Pull request by mkurz**
Same like #10121, however let's also add this for the JSON error handlers (and for Scala's http error handler, which I forgot in the original pull request).

### CODE REVIEW

File: core/play/src/main/scala/play/api/http/HttpErrorHandler.scala [hunk 1]
```diff
@@ -259,10 +259,26 @@ class DefaultHttpErrorHandler(
...
+   * simple string, without doing any fancy processing inside the method (like accessing files,...)
+   * that could throw exceptions. This is your last chance to send a meaningful error message when
+   * everything else failed.
+   *
+   * @param request   The request that triggered the server error.
+   * @param exception The server error.
+   * @return An error message which will be send as last resort in case handling a server error with
+   *         this error handler failed.
+   */
+  protected def fatalErrorMessage(request: RequestHeader, exception: Throwable) = ""
```
Comment (PromanSEW) [inline core/play/src/main/scala/play/api/http/HttpErrorHandler.scala:280]: Method return type is missing: `String`?

Comment (mkurz) [reply core/play/src/main/scala/play/api/http/HttpErrorHandler.scala:280]: Is it necessary? The code compiles, no complains.

Comment (gmethvin) [reply core/play/src/main/scala/play/api/http/HttpErrorHandler.scala:280]: It's not strictly necessary, but it's the accepted convention pretty much in Scala, especially when we care about the public API and binary compatibility concerns. This is a trivial case, but I've definitely seen problems before where we meant the API to expose a different type than what was actually inferred by the compiler.

Comment (mkurz) [reply core/play/src/main/scala/play/api/http/HttpErrorHandler.scala]: Return type added.

File: core/play/src/main/scala/play/api/http/HttpErrorHandler.scala [hunk 2]
```diff
@@ -404,9 +420,25 @@ class JsonHttpErrorHandler(environment: Environment, sourceMapper: Option[Source
...
+   * simple JsonNode, without doing any fancy processing inside the method (like accessing files,...)
+   * that could throw exceptions. This is your last chance to send a meaningful error message when
+   * everything else failed.
+   *
+   * @param request   The request that triggered the server error.
+   * @param exception The server error.
+   * @return An error JSON which will be send as last resort in case handling a server error with
+   *         this error handler failed.
+   */
+  protected def fatalErrorMessage(request: RequestHeader, exception: Throwable): JsValue = Json.obj()
```
Comment (gmethvin) [inline core/play/src/main/scala/play/api/http/HttpErrorHandler.scala:440]: Maybe the name should be `fatalErrorJson` or something like that? `Message` seems to imply this should just be a string, whereas this is the full JSON content.

Comment (mkurz) [reply core/play/src/main/scala/play/api/http/HttpErrorHandler.scala]: Changed the name to `fatalErrorJson`, makes sense.

**Comment by mkurz** [comment]
@Mergifyio backport 2.8.x

**Pull request merged**